From 254ba478ac31c5671d723520664ae38de3e0600c Mon Sep 17 00:00:00 2001
From: Andrei V. FOMITCHEV <andreix.fomitchev@intel.com>
Date: Fri, 8 Jun 2012 17:55:08 +0200
Subject: [PORT FROM MAIN] accoustics effects process

BZ: 31726

The purpose of this patch is to propagate the errors of WebRTC to upper layers

Feature: AudioComms, WebRTC
Category: Integration
Purpose: PV-Enabler
Origin: Internal
Upstream-candidate: yes
Change-Id: Ife2c1aebbeff90342fbf41bdcd9fa719cf08d47d
Orig-Change-Id: I87c6812b621768f04b4ad80a38fa5604ce97c7ed
Signed-off-by: Andrei V. FOMITCHEV <andreix.fomitchev@intel.com>
Reviewed-on: http://android.intel.com:8080/76840
Reviewed-by: Barthes, FabienX <fabienx.barthes@intel.com>
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libeffects/preprocessing/PreProcessing.cpp |   13 ++++++-------
 1 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/media/libeffects/preprocessing/PreProcessing.cpp b/media/libeffects/preprocessing/PreProcessing.cpp
index 25586e8..31f340d 100644
--- a/media/libeffects/preprocessing/PreProcessing.cpp
+++ b/media/libeffects/preprocessing/PreProcessing.cpp
@@ -1281,7 +1281,7 @@ int PreProcessingFx_Process(effect_handle_t     self,
         session->procFrame->_payloadDataLengthInSamples =
                 session->apmFrameCount * session->inChannelCount;
 
-        effect->session->apm->ProcessStream(session->procFrame);
+        status = effect->session->apm->ProcessStream(session->procFrame);
 
         if (session->outBufSize < session->framesOut + session->frameCount) {
             session->outBufSize = session->framesOut + session->frameCount;
@@ -1325,11 +1325,10 @@ int PreProcessingFx_Process(effect_handle_t     self,
               (session->framesOut - fr) * session->outChannelCount * sizeof(int16_t));
         session->framesOut -= fr;
         outBuffer->frameCount += fr;
-
-        return 0;
     } else {
-        return -ENODATA;
+        status = -ENODATA;
     }
+    return status;
 }
 
 int PreProcessingFx_Command(effect_handle_t  self,
@@ -1791,11 +1790,11 @@ int PreProcessingFx_ProcessReverse(effect_handle_t     self,
         }
         session->revFrame->_payloadDataLengthInSamples =
                 session->apmFrameCount * session->inChannelCount;
-        effect->session->apm->AnalyzeReverseStream(session->revFrame);
-        return 0;
+        status = effect->session->apm->AnalyzeReverseStream(session->revFrame);
     } else {
-        return -ENODATA;
+        status = -ENODATA;
     }
+    return status;
 }
 
 
-- 
1.7.4.1

